lib/config.js

```diff
@@ -229,6 +230,16 @@ function getLocalConfig(thisConfig, directory) {
             continue;
         }
 
+        // Check for root flag
+        if (localConfig.root === true) {
+            rootPath = path.dirname(localConfigFile);
+        }
+
+        // If root flag is set, don't consider file if it is above root
+        if (rootPath && localConfigFile.indexOf(rootPath) === -1) {
```

Comment:
With my limited experience, I don't know what else to use, unless you'd prefer I add a dependency for `path-is-inside` as @gyandeeps suggests.

Or perhaps this would be safer? (only if the root folder is named `.eslintrc` or `package.json` I think...)

```
if (rootPath && path.dirname(localConfigFile).indexOf(rootPath) === -1) {
```